src/components/Accordion/Accordion.tsx

```diff
@@ -126,18 +150,33 @@ export const Accordion: FC<AccordionProps> = (props) => {
     const {
         // @react-aria prevents default action for onMouseDown as implemented here: https://github.com/adobe/react-spectrum/blob/e14523fedd93ac1a4ede355aed70988af572ae74/packages/%40react-aria/selection/src/useSelectableCollection.ts#L370
         // This makes it impossible to edit or select text in input fields inside the accordion
+        // onKeydown automatically navigates away from child elements, including dropdown menus etc.
+        // Focus should only be passed if the accordion item itself is focused
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        accordionProps: { onMouseDown, ...accordionProps },
+        accordionProps: { onMouseDown, onKeyDown, ...accordionProps },
         // @react-aria enable by default typeahead which result in an event fired up on keypress and select the section
         // eslint-disable-next-line @typescript-eslint/ban-ts-comment
         //@ts-ignore
     } = useAccordion({ ...ariaProps, disallowTypeAhead: true }, state, ref);
 
+    const accordionKeyDown = (event: KeyboardEvent<HTMLDivElement>) => {
+        const key = (event.target as HTMLButtonElement).dataset.key;
+        const isFocused = state.selectionManager.focusedKey === key;
+        if (key && isFocused) {
+            onKeyDown && onKeyDown(event);
+        } else if (ACCORDION_CONTROL_KEYS.includes(event.key)) {
```

Comment:
could you try to use https://react-spectrum.adobe.com/react-aria/useFocusWithin.html instead, so you don't need to deal with control key checks?

Reply:
fixed as discussed :)